docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -1078,6 +1079,122 @@ public @interface CanWrite {
 ----
 <1> Any method or class annotated with the `@CanWrite` annotation is secured with this `@PermissionsAllowed` annotation instance.
 
+[[permission-bean-params]]
+=== Pass `@BeanParam` parameters into a custom permission
+
+Quarkus can map fields of a secured method parameters to a custom permission constructor parameters.
+You can use this feature to pass `jakarta.ws.rs.BeanParam` parameters into your custom permission.
+Let's consider following Jakarta REST resource:
+
+[source,java]
+----
+package org.acme.security.rest.resource;
+
+import io.quarkus.security.PermissionsAllowed;
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+@Path("/hello")
+public class SimpleResource {
+
+    @PermissionsAllowed(value = "read", permission = BeanParamPermission.class,
+        params = "beanParam.securityContext.userPrincipal.name") <1>
```

Comment:
It looks like you are supporting unlimited unwrapping of BeanParam params, but does that make it too complex ? I thought the plan was only to allow to refer to individual BeanParam params.

Reply:
Implementation wise, it was just difference few recursive calls, so I thought it was useful. I can drop it either from documentation or from implementation itself.  Personally, I don't see an appeal to give ability to reference `beanParam.query` and not `beanParam.nestedBeanParam.query`. For example here https://github.com/quarkusio/quarkus/blob/96d896763a6e2b7f862bf860b1b36f16f0403c64/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/BeanParamRecordTest.java#L71

I checked out tests, saw how `@ BeanParam` are tested and tried to support that. My personal opinion is that it is cool that I can just reference principal directly.

Reply:
Yeah agreed. But getting awfull close to EL 😱